plugins/net_plugin/net_plugin.cpp

```diff
@@ -790,8 +983,22 @@ namespace eosio {
       void set_connection_type( const string& peer_addr );
       bool is_transactions_only_connection()const { return connection_type == transactions_only; } // thread safe, atomic
       bool is_blocks_only_connection()const { return connection_type == blocks_only; }
-      bool is_transactions_connection() const { return connection_type != blocks_only; } // thread safe, atomic
-      bool is_blocks_connection() const { return connection_type != transactions_only; } // thread safe, atomic
+
+      //add more connection type
+      bool is_peers_only_connection() const { return connection_type == peers_only; }
```

Comment:
I don't understand a peers only connection. You run a node that does nothing more than share out peers it is connected to? It does not forward blocks or transactions?

Reply:
We can set up a service node dedicated to P2P exchanges, especially within a large-scale network.